core-library/src/main/java/org/silverpeas/core/admin/persistence/GroupUserRoleTable.java

```diff
@@ -116,81 +119,84 @@ protected void prepareInsert(String insertQuery, PreparedStatement insert, Group
   @Override
   protected void prepareUpdate(String updateQuery, PreparedStatement update, GroupUserRoleRow row)
       throws SQLException {
+    // nothing to do here
   }
 
   /**
    * Delete the groupUserRole
    */
-  public void removeGroupUserRole(int id) throws AdminPersistenceException {
+  public void removeGroupUserRole(int id) throws SQLException {
     GroupUserRoleRow groupUserRole = getGroupUserRole(id);
     if (groupUserRole == null) {
       return;
     }
 
-    UserRow[] users = organization.user.getDirectUsersOfGroupUserRole(id);
-    for (UserRow user : users) {
-      removeUserFromGroupUserRole(user.id, id);
-    }
+    try (Connection connection = DBUtil.openConnection()) {
+      List<String> userIds =
+          userDAO.getDirectUserIdsByGroupUserRole(connection, String.valueOf(id));
+      for (String userId : userIds) {
+        removeUserFromGroupUserRole(Integer.parseInt(userId), id);
+      }
 
-    GroupRow[] groups = organization.group.getDirectGroupsInGroupUserRole(id);
-    for (GroupRow group : groups) {
-      removeGroupFromGroupUserRole(group.id, id);
+      List<String> groupIds =
+          groupDAO.getDirectGroupIdsByGroupUserRole(connection, String.valueOf(id));
+      for (String groupId : groupIds) {
+        removeGroupFromGroupUserRole(Integer.valueOf(groupId), id);
+      }
+
+
+      updateRelation(DELETE_GROUPUSERROLE, id);
     }
 
-    // organization.userSet.removeUserSet("H", id);
-    updateRelation(DELETE_GROUPUSERROLE, id);
   }
 
-  static final private String DELETE_GROUPUSERROLE = "delete from ST_GroupUserRole where id = ?";
+  private static final String DELETE_GROUPUSERROLE = "delete from ST_GroupUserRole where id = ?";
 
   /**
    * Tests if a user has a given role (not recursive).
    */
   private boolean isUserDirectlyInRole(int userId, int groupUserRoleId)
-      throws AdminPersistenceException {
+      throws SQLException {
     int[] ids = new int[] { userId, groupUserRoleId };
     Integer result = getInteger(SELECT_COUNT_GROUPUSERROLE_USER_REL, ids);
 
     return result != null && result >= 1;
   }
 
-  static final private String SELECT_COUNT_GROUPUSERROLE_USER_REL =
+  private static final String SELECT_COUNT_GROUPUSERROLE_USER_REL =
       "select count(*) from ST_GroupUserRole_User_Rel"
       + " where userId = ? and groupUserRoleId = ?";
 
   /**
    * Add an user in a groupUserRole.
    */
   public void addUserInGroupUserRole(int userId, int groupUserRoleId) throws
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Refactor this method to throw at most one checked exception instead of: java.sql.SQLException, org.silverpeas.core.admin.service.AdminException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1160)